board/drivers/fdcan.h

```diff
@@ -0,0 +1,195 @@
+// IRQs: FDCAN1_IT0, FDCAN1_IT1
+//       FDCAN2_IT0, FDCAN2_IT1
+//       FDCAN3_IT0, FDCAN3_IT1
+
+FDCAN_GlobalTypeDef *cans[] = {FDCAN1, FDCAN2, FDCAN3};
+
+bool can_set_speed(uint8_t can_number) {
+  bool ret = true;
+  FDCAN_GlobalTypeDef *CANx = CANIF_FROM_CAN_NUM(can_number);
+  uint8_t bus_number = BUS_NUM_FROM_CAN_NUM(can_number);
+
+  ret &= llcan_set_speed(CANx, can_speed[bus_number], can_data_speed[bus_number], can_loopback, (unsigned int)(can_silent) & (1U << can_number));
+  return ret;
+}
+
+void can_set_gmlan(uint8_t bus) {
+  UNUSED(bus);
+  puts("GMLAN not available on red panda\n");
+}
+
+void cycle_transceivers(void) {
+  current_board->enable_can_transceiver(1, false);
```

Comment:
let's look into this more, don't think we need this

Reply:
discussed, this is here because CANFD doesn't init until there is 129 recessive bits in a row. Toggling the transciever is a way to make that happen.

Think we want to do this instead:
- Try initting normally first. if that doesn't happen in x amount of time, cycle the transciever
- Only cycle the transciever which is giving issues
- Print something when this triggers for debugging purpose